jax/_src/lax/control_flow.py

```diff
@@ -336,7 +336,7 @@ def _while_loop_translation_rule(c, axis_env, name_stack, avals, backend, *args,
   x, _, z = split_list(cond_carry_elts, [cond_nconsts, body_nconsts])
   pred, = xla.jaxpr_subcomp(cond_c, cond_jaxpr.jaxpr, backend, axis_env,
                             _map(partial(xb.constant, cond_c), cond_jaxpr.consts),
-                            extend_name_stack(name_stack, 'cond'), *(x + z))
+                            name_stack.extend(while_p, 'cond'), *(x + z))
```

Comment:
As I am reading through this linearly, I notice that this is one case where `extend_name_stack` was called without `xla.wrap_name`. This means that we are changing the way the XLA op_names will look. Is `while(cond)` better than `while/cond`? Perhaps.

Reply:
My original choice to use `while/cond` and `while/body` was pretty arbitrary, and I don't think it's load-bearing.